File: src/brg-utst/test-memctrl/TestMemCtrlHarness.py
from TestHarness import TestHarness
from m5.params import *
from m5.proxy import *

class TestMemCtrlHarness(TestHarness):
  type = 'TestMemCtrlHarness'
  cxx_header = "brg-utst/test-memctrl/test_memctrl_harness.hh"

File: src/brg-utst/test-memctrl/test_memctrl_harness.cc
//========================================================================
// Test Memory Controller Harness
//========================================================================
//
// Authors  : Tuan Ta
// Date     : Sep 26, 2018
//

#include "test_memctrl_harness.hh"
#include "test_memctrl_suite.hh"

TestMemCtrlHarness::TestMemCtrlHarness(const Params* params)
    : TestHarness(params)
{
  test_suite_ptr = new TestMemCtrlSuite(num_test_ports);

  // init directed tests
  test_suite_ptr->initDirectedTests();
}

TestMemCtrlHarness*
TestMemCtrlHarnessParams::create()
{
  return new TestMemCtrlHarness(this);
}
